src/distilabel/pipeline.py

```diff
@@ -439,14 +504,20 @@ def _build_dataset(  # noqa: C901
         _dataset.task = self.labeller.task if self.labeller is not None else None  # type: ignore
         return _dataset  # type: ignore
 
+    def _teardown(self) -> None:
+        if self.generator is not None and isinstance(self.generator, ProcessLLM):
+            self.generator.teardown()
+
+        if self.labeller is not None and isinstance(self.labeller, ProcessLLM):
+            self.labeller.teardown()
+
     def _generate(  # noqa: C901
         self,
         dataset: Dataset,
         num_generations: int = 1,
         batch_size: int = 1,
         enable_checkpoints: bool = True,
```

Comment:
IMO this could be named in a better way, mentioning in case you have ideas for an upcoming PR before the release i.e. `fail_on_error=False` or something similar, as we do return something similar to a checkpoint instead of raising and error, but we don't to checkpointing every N samples, so maybe `enable_checkpoints` is misleading (my bad on naming again :/)

Reply:
Yeah, I think using `checkpoints` is missleading, because is not a checkpoint, is just saving the current progress in the dataset if the generation process fails.